sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1996,6 +1998,22 @@ def provision_with_retries(
                                        skip_unnecessary_provisioning else None)
 
         failover_history: List[Exception] = list()
+        # When jobs controller/server using the local credentials which are
+        # expiring it may cause the cluster to be leaked. So, checking the
+        # enabled clouds and expiring credentials and warning the user to use
+        # the credentials that never expire or a service account.
+        if task.is_controller_task():
+            enabled_clouds = sky_check.get_cached_enabled_clouds_or_refresh()
+            expirable_clouds = backend_utils.get_expirable_clouds(
+                enabled_clouds)
+
+            if len(expirable_clouds) > 0:
+                warnings = (f'\nWarning: Expiring credentials detected for '
+                            f'{expirable_clouds}. Clusters may be leaked if '
+                            f'the credentials expire while jobs are running. '
+                            f'It is recommended to use credentials that never'
+                            f' expire or a service account.')
```

Comment:
Since we do not verify if the credentials are expiring, we should update the message:
```suggestion
                warnings = (f'\nWarning: Credentials used for {expirable_clouds} may expire. '
                            f' Resources may be leaked if '
                            f'the credentials expire while jobs are running. '
                            f'It is recommended to use credentials that never'
                            f' expire or a service account.')
```

Reply:
updated